Service/EnrichPubliccodeService.php

```diff
@@ -0,0 +1,208 @@
+    private Entity $componentEntity;
+    private Entity $repositoryEntity;
+    private Source $githubApi;
+
+    public function __construct(
+        EntityManagerInterface $entityManager,
+        CallService $callService
+    ) {
+        $this->callService = $callService;
+        $this->entityManager = $entityManager;
+
+        $this->configuration = [];
+        $this->data = [];
+    }
+
+    /**
+     * Set symfony style in order to output to the console
+     *
+     * @param SymfonyStyle $io
+     * @return self
+     */
+    public function setStyle(SymfonyStyle $io): self
+    {
+        $this->io = $io;
+
+        return $this;
+    }
+
+    /**
+     * Updates component and repository with publiccode.yml data
+     * 
+     * @param ObjectEntity $repository
+     * @param array        $publiccode
+     *
+     * @return ObjectEntity|null dataset at the end of the handler
+     */
+    public function enrichRepositoryWithPubliccode(ObjectEntity $repository, array $publiccode = []): ?ObjectEntity
+    {
+        if (!$component = $repository->getValue('component')) {
+            $component = new ObjectEntity($this->componentEntity);
+        }
+
+        if ($publiccode !== null) {
+            if (!$description = $component->getValue('description')) {
+                $description = new ObjectEntity($this->descriptionEntity);
+            }
+
+            $description->hydrate([
+                'shortDescription' => $publiccode['description']['en']['shortDescription'] ?? $publiccode['description']['nl']['shortDescription'] ?? null,
+                'documentation' => $publiccode['description']['en']['documentation'] ?? $publiccode['description']['nl']['documentation'] ?? null,
+                'apiDocumentation' => $publiccode['description']['en']['apiDocumentation'] ?? $publiccode['description']['nl']['apiDocumentation'] ?? null,
+            ]);
+            $this->entityManager->persist($description);
+
+            $component->hydrate([
+                'softwareVersion' => $publiccode['publiccodeYmlVersion'] ?? null,
+                'name' => $publiccode['name'] ?? null,
+                'softwareType' => $publiccode['softwareType'] ?? null,
+                'inputTypes' => $publiccode['inputTypes'] ?? null,
+                'outputTypes' => $publiccode['outputTypes'] ?? null,
+                'platforms' => $publiccode['platforms'] ?? null,
+                'categories' => $publiccode['categories'] ?? null,
+                'developmentStatus' => $publiccode['developmentStatus'] ?? null,
+                'url' => $repository->getId()->toString() ?? null,
+                'description' => $component->getId()->toString(),
+                // Disabled because ? 
+                // 'releaseDate' => $publiccode['releaseDate'] ?? null,
+                // 'dependsOn' => $publiccode['dependsOn']['open']['name'] ?? null,
+                // 'dependsOn' => $publiccode['dependsOn']['open']['versionMin'] ?? null,
```

Comment:
Because it gives errors, or because it was already disabled in a previous version?

Reply:
unkown